bson/bsonoptions/struct_codec_options.go

```diff
@@ -38,6 +39,12 @@ func (t *StructCodecOptions) SetEncodeOmitDefaultStruct(b bool) *StructCodecOpti
 	return t
 }
 
+// SetAllowOverwritingEmbeddedFields specifies if embedded fields can be overwritten by higher level fields. Defaults to true.
```

Comment:
I think an example here may be helpful (if that is appropriate in Go documentation for individual functions).

IIUC the default is actually false. We will overwrite the higher level field with the value of the embedded field.

Reply:
I'm not adding an example is a thing, but i tried to explain in more depth.
Unless I'm mistaking what higher level means, this should be correct? the inner inlined struct field gets overwritten by the field on the outside struct

Reply:
The organization is a little confusing, but the default registry encoders and decoders are set in RegisterDefaultDecoders and RegisterDefaultEncoders in bsoncodec in default_value_decoders.go and default_value_encoders.go.  The NewRegistryBuilder I'm changing in this change is the mgocompat one.

Reply:
Ah I missed that. Thanks for the explanation!